src/Google.Api.Gax/PagedAsyncEnumerable.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Protobuf;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Api.Gax
+{
+    public sealed class PagedAsyncEnumerable<TRequest, TResponse, TResource> : IPagedAsyncEnumerable<TResponse, TResource>
+        where TRequest : class, IPageRequest, IMessage<TRequest>
+        where TResponse : class, IPageResponse<TResource>, IMessage<TResponse>
+    {
+        private readonly CallSettings _callSettings;
+        private readonly TRequest _request;
+        private readonly ApiCall<TRequest, TResponse> _apiCall;
+
+        public PagedAsyncEnumerable(CallSettings callSettings,
```

Comment:
Reorder the parameters? It feels a little odd for the call settings to be before the request.